src/mappings/DAOTracker/mapping.ts

```diff
@@ -0,0 +1,146 @@
+import { Address, ipfs, store } from '@graphprotocol/graph-ts';
+import {
+  BlacklistDAO,
+  DAOTracker,
+  ResetDAO,
+  TrackDAO,
+} from '../../types/DAOTracker/DAOTracker';
+import {
+  AvatarContract,
+  BlacklistedDAO,
+  ContractInfo,
+  DAOTrackerContract,
+  ResetDAO as ResetDAOEntity,
+  UControllerOrganization,
+} from '../../types/schema';
+import { createTemplate, equalStrings, fetchTemplateName } from '../../utils';
+
+export function getDAOTrackerContract(address: Address): DAOTrackerContract {
+  let daoTracker = DAOTrackerContract.load(address.toHex()) as DAOTrackerContract;
+  if (daoTracker == null) {
+    daoTracker = new DAOTrackerContract(address.toHex());
+    daoTracker.address = address;
+    let daoTrackerSC = DAOTracker.bind(address);
+    daoTracker.owner = daoTrackerSC.owner();
+    daoTracker.save();
+  }
+  return daoTracker;
+}
+
+export function handleTrackDAO(event: TrackDAO): void {
+  // Ensure the DAOTrackerContract has been added to the store
+  getDAOTrackerContract(event.address);
+
+  let avatar = event.params._avatar;
+  let controller = event.params._controller;
+  let reputation = event.params._reputation;
+  let daoToken = event.params._daoToken;
+  let sender = event.params._sender;
+  let arcVersion = event.params._arcVersion;
+
+  // If the avatar already exists, early out
+  if (AvatarContract.load(avatar.toHex()) != null) {
+    return;
+  }
+
+  // If the sender of the 'track' call is the DaoCreator contract, use its arcVersion
+  let daoCreatorInfo = ContractInfo.load(sender.toHex());
+  if (daoCreatorInfo != null) {
+    if (equalStrings(daoCreatorInfo.name, 'DaoCreator')) {
+      arcVersion = daoCreatorInfo.version;
+    }
+  }
+
```

Comment:
need to confirm (somehow) that creating a template for wrong contract does not crash the subgraph .
this can happened only if the event does not come from daocreator.
if we cannot confirm ,so lets consider  limit this feature only for daocreator.

Reply:
The subgraph does not (and cannot) check to make sure the contract bytecode matches the bytecode in the ABI file. This means that it only watches for events ABIs from a given contract address, and if they do not emit from that address the subgraph will not do anything.

So in the case you're describing, the subgraph would not crash when it creates the template, it would rather just do nothing from there on out. The only case where this could happen is someone wrote custom contracts that emit the events we're looking for, and the subgraph responds to those events from these newly created contracts. The plan here, is that if DAOstack detects a "DAO" is doing this, it would blacklist it.

Reply:
Writing a test to verify the subgraph doesn't crash.

Reply:
as far as I see the only way to mitigate that and stay on the safe side is to track this event only if it come via daocreator. otherwise ignore it.